bitpay/bitpay-app
# Feat/upgrade segment sdk

This PR upgrades the Segment SDK from `1.5.2` => `2.7.1`. AppsFlyer is now device-mode instead of cloud-mode (per discussion with AppsFlyer, cloud-mode not working/not supported at this time). Created a wrapper lib for Segment SDK client to have better control over initialization and guard functions.

Testing:
- confirm that Segment is correctly receiving events on iOS/Android and that the advertising ID is being passed (iOS). (the Segment dashboard debugger shows real-time events)
- confirm that AppsFlyer is correctly receiving events (this is kinda tricky... you may have to do a build with a button or something that calls `track('Internal Dev Test Event', {})`, then wait and see if the count increments on the AppsFlyer dashboard, it can take around an hour or so for AppsFlyer dashboard to update)

Added:
- `@segment/analytics-react-native` `2.7.1` - latest SDK
- `@segment/sovran-react-native` `0.4.4` - required by Segment SDK 2.x.x
- `@segment/analytics-react-native-plugin-idfa` `0.4.1`- replace old IDFA initialization
- `@segment/analytics-react-native-plugin-appsflyer` `0.4.0` - replace manual AppsFlyer integration

Upgraded:
- `@react-native-async-storage/async-storage` `1.15.14` => `1.17.10` - suggested by Segment SDK 2.x.x

Removed:
- `react-native-advertising-id-bp` `1.0.13` - this an was in-house custom fork of another dependency needed to get the Advertising ID to pass to Segment. Replaced with Segment's own IDFA plugin.
